### PR TITLE
[Issue 14052] Fixes use default KEY_SHARE ``ConsistentHashingStickyKeyConsumerSelector`` cause flaky-test

### LINKED ISSUE #14052: Flaky-test: PerformanceProducerTest.testMsgKey
[org.apache.pulsar.testclient.PerformanceProducerTest] is flaky. The [testMsgKey] test method fails sporadically.

[example failure](https://github.com/apache/pulsar/runs/4991734305?check_suite_focus=true#step:8:27104)

```
Error:  testMsgKey(org.apache.pulsar.testclient.PerformanceProducerTest)  Time elapsed: 1.155 s  <<< FAILURE!
java.lang.AssertionError: expected [true] but found [false]
        at org.testng.Assert.fail(Assert.java:99)
        at org.testng.Assert.failNotEquals(Assert.java:1037)
        at org.testng.Assert.assertTrue(Assert.java:45)
        at org.testng.Assert.assertTrue(Assert.java:55)
        at org.apache.pulsar.testclient.PerformanceProducerTest.testMsgKey(PerformanceProducerTest.java:112)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:566)
        at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:132)
        at org.testng.internal.InvokeMethodRunnable.runOne(InvokeMethodRunnable.java:45)
        at org.testng.internal.InvokeMethodRunnable.call(InvokeMethodRunnable.java:73)
        at org.testng.internal.InvokeMethodRunnable.call(InvokeMethodRunnable.java:11)
        at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
        at java.base/java.lang.Thread.run(Thread.java:829)
```

### PR DESCRIPTION
**Pull request by mattisonchao**
Fixes #14052

Master Issue: #14052

### Motivation

See #14052

### Modifications

- Make producer start after the consumer.
- Make random message key distribute eventually consistent.

### Verifying this change

- [x] Make sure that the change passes the CI checks.

### Documentation

- [x] `no-need-doc`

### CODE REVIEW

**Comment by mattisonchao** [comment]
@lhotari  @Jason918  PTAL :)

File: pulsar-testclient/src/main/java/org/apache/pulsar/testclient/PerformanceProducer.java [hunk 1]
```diff
@@ -736,7 +736,7 @@ private static void runProducer(int producerId,
                     }
                     //generate msg key
                     if (msgKeyMode == MessageKeyGenerationMode.random) {
-                        messageBuilder.key(String.valueOf(random.nextInt()));
+                        messageBuilder.key(String.valueOf(random.nextInt(Integer.MAX_VALUE)));
```
Comment (Jason918) [inline pulsar-testclient/src/main/java/org/apache/pulsar/testclient/PerformanceProducer.java:739]: Can you explain some details about this change?

File: pulsar-testclient/src/test/java/org/apache/pulsar/testclient/PerformanceProducerTest.java [hunk 2]
```diff
@@ -119,7 +120,7 @@ public void testMsgKey() throws Exception {
         }
 
         //use msg key generator,so every consumer can get msg
-        String newArgString = "%s -r 10 -u %s -m 500 -mk autoIncrement";
+        String newArgString = "%s -r 10 -u %s -m 500 -mk random";
```
Comment (Jason918) [inline pulsar-testclient/src/test/java/org/apache/pulsar/testclient/PerformanceProducerTest.java:123]: What's wrong with autoIncrement?

**Review by Jason918 — APPROVED** [review]
Overall LGTM.
Left some questions.

**Comment by mattisonchao** [comment]
Hi, @Jason918 

I saw your comment and thought maybe we don't need to make changes.
So I will roll back the changes you left in the comments.

**Comment by mattisonchao** [comment]
@Jason918 

I found out that I pushed a bad commit to this PR and I have now cleaned it up.

PTAL :-)

**Review by Jason918 — APPROVED** [review]
+1

**Review by merlimat — APPROVED** [review]

**Pull request merged**